vernac/declare.ml

```diff
@@ -1747,7 +1765,8 @@ let close_proof_delayed ~feedback_id ps (fpl : closed_proof_output Future.comput
   let entries = CList.map_i make_entry 0 (Proofview.initial_goals entry) in
   { name; entries; uctx = initial_euctx; pinfo }
 
-let close_future_proof = close_proof_delayed
+let stm_close_future_proof = stm_close_proof_delayed
+let stm_close_proof = close_proof ~leak_seff:false
```

Comment:
Since the stm rewinds the state before Qed/Defined, side effects don't leak

Reply:
this seems like a comment that should be in the code